pkg/strategy/translation/translation.go

```diff
@@ -64,8 +65,21 @@ func (t *Strategy) toPublicObjects(ctx context.Context, objs ...runtime.Object)
 
 func (t *Strategy) toPublic(ctx context.Context, obj runtime.Object, err error, namespace, name string) (types.Object, error) {
 	if err != nil {
+		// if err is a not found error, translate its Kind back to the public version
+		if apierrors.IsNotFound(err) {
+			gvk, err := apiutil.GVKForObject(t.translator.NewPublic(), t.strategy.Scheme())
+			if err != nil {
+				return nil, err
+			}
+
+			return nil, apierrors.NewNotFound(schema.GroupResource{
+				Group:    gvk.Group,
+				Resource: gvk.Kind,
```

Comment:
`Kind` will have capital letters and will be singular.

`Resource` should be all lowercase and be plural.

Reply:
That's true, but I don't know of a way to change this Kind into a Resource without having to add another function to the Translator interface, which would then require changes in every single application that uses mink (or at least, every application that uses mink and has both public and internal types).

Here's a few examples of what the errors look like with this change:

```
❯ acorn app dne
  ✗  ERROR:  App.api.acorn.io "dne" not found

❯ acorn container dne
  ✗  ERROR:  ContainerReplica.api.acorn.io "dne" not found

❯ acorn credential dne
  ✗  ERROR:  Credential.api.acorn.io "dne" not found
```

If we want, I can add some kind of function like `PublicGroupResource` to the Translator interface so that we can get that `schema.GroupResource` object. If you know of a way that we could reliably convert this Kind to a Resource here in the mink code, let me know, because I could not think of or discover one.

Reply:
Acorn uses `strings.ToLower(name.GuessPluralName(c.GVK.Kind))`, where the `name.GuessPluralname` comes from `github.com/rancher/wrangler`.

Reply:
Perfect, thanks! I'll go try that out.